tests/Collection.test.ts

```diff
@@ -177,4 +183,122 @@ describe("Collection", () => {
+          reader.readAsText(file);
+        });
+        return new Response(fileContents, {
+          status: 200,
+          // eslint-disable-next-line
+          headers: { "Content-Type": file.type },
+        });
+      }
+      return new Response("Not Found", {
+        status: 404,
+      });
+    }
+
+    //// Constants ////
+
+    const MOCK_COLLECTION_JSON = `[{"path": "dataset", "name": "dataset"}]`;
+    const MOCK_COLLECTION_FILEMAP = {
+      // eslint-disable-next-line
+      "collection.json": new File([MOCK_COLLECTION_JSON], "collection.json"),
+    };
+    const MOCK_DATASET_FILEMAP = {
+      // eslint-disable-next-line
+      "manifest.json": new File([JSON.stringify(MOCK_DATASET_MANIFEST)], "manifest.json"),
+      // add minimal files to ensure the Dataset can actually load
+      // eslint-disable-next-line
+      "times.json": new File([JSON.stringify(MOCK_DATASET_TIMES)], "times.json"),
+      // eslint-disable-next-line
+      "feature1.json": new File([JSON.stringify(MOCK_DATASET_FEATURE_1)], "frames.json"),
+    };
+
+    const DATASET_LOAD_OPTIONS: DatasetLoadOptions = {
+      manifestLoader: async (url) => {
+        return await mockFetch(url).then((response: Response) => response.json());
+      },
+      arrayLoader: new MockFetchArrayLoader(mockFetch),
+    };
+    const COLLECTION_LOAD_CONFIG: CollectionLoadOptions = {
+      fetchMethod: mockFetch,
+      datasetLoadOptions: DATASET_LOAD_OPTIONS,
+    };
+
+    //// Tests ////
+
+    it("can load a collection from a file map", async () => {
+      const collection = await Collection.loadFromAmbiguousFile("", MOCK_COLLECTION_FILEMAP, COLLECTION_LOAD_CONFIG);
+
+      expect(collection).to.be.instanceOf(Collection);
+      expect(collection.getDatasetKeys().length).to.equal(1);
+      expect(collection.hasDataset("dataset")).to.be.true;
+      expect(collection.getAbsoluteDatasetPath("dataset")).to.equal("dataset/manifest.json");
+      expect(collection.sourcePath).not.to.be.null;
+      expect(collection.sourceType).toEqual(CollectionSourceType.ZIP_FILE);
+      collection.dispose();
+    });
+
+    it("can load a single dataset from a file map", async () => {
+      const collection = await Collection.loadFromAmbiguousFile("", MOCK_DATASET_FILEMAP, COLLECTION_LOAD_CONFIG);
+
+      expect(collection).to.be.instanceOf(Collection);
+      expect(collection.getDatasetKeys().length).to.equal(1);
+      expect(collection.hasDataset("manifest.json")).to.be.true;
+      expect(collection.getAbsoluteDatasetPath("manifest.json")).to.equal("manifest.json");
+      expect(collection.sourcePath).to.be.null;
+      expect(collection.sourceType).toEqual(CollectionSourceType.ZIP_FILE);
+      collection.dispose();
+    });
+
+    it("throws an error if no manifest files exist", async () => {
+      const fileMap = {};
+      await expect(Collection.loadFromAmbiguousFile("", fileMap, COLLECTION_LOAD_CONFIG)).rejects.toThrowError(
+        ANY_ERROR
```

Comment:
this feels a little nit-picky, but since you have errors defined can't you just put the specific error here (and below) you expect?